### PR TITLE
fix(documentation): Remove mentions of legacy extend

### PR DESCRIPTION
**Pull request by mrskycriper**
- Remove usages of legacy extend
- Makes more emphasis on modern extends for new device support

### CODE REVIEW

File: docs/advanced/support-new-devices/01_support_new_devices.md [hunk 1]
```diff
@@ -25,12 +25,21 @@ Make sure that joining is enabled, otherwise new devices cannot join the network
...
-Next generate the external definition by going to the device -> "Dev console" tab and press "Generate external definition". 
-If all features work and all the expected features are there, you are lucky and can continue with step 3.
-If not, we continue by extending the external definition.
+Next generate the external definition by going to the device -> "Dev console" tab and press "Generate external definition".
 
 <img src="../../images/generate_external_definition.gif" height="300"/>
 
+By default, the external definition will use modern extends.
+This is done by mapping them to exposed Zigbee clusters.
+But this approach only works well when devices implement the Zigbee specification.
```
Comment (Koenkk) [inline docs/advanced/support-new-devices/01_support_new_devices.md:34]:
```suggestion
But this approach only works well when devices implement the Zigbee ZCL specification.
```

File: docs/advanced/support-new-devices/01_support_new_devices.md [hunk 2]
```diff
@@ -134,6 +137,30 @@ Repeat until your device does not produce any more log messages like: `2018-5-1
 
 If none of the existing converters fit you can add custom ones, an external converter example for this can be found [here](https://github.com/Koenkk/zigbee2mqtt.io/blob/master/docs/externalConvertersExample/freepad_ext.js).
 
+### 2.3 Using modern extend
```
Comment (Koenkk) [inline docs/advanced/support-new-devices/01_support_new_devices.md:140]: I'm wondering if this should replace 2.2 completely.

Comment (mrskycriper) [reply docs/advanced/support-new-devices/01_support_new_devices.md:140]: Probably, but I will take some time to make a better guide. So I added this as a temporary fix for now.

**Comment by Koenkk** [comment]
Thanks!

**Pull request merged**